docs/decisions/0001-adding-new-checks.rst

```diff
@@ -0,0 +1,36 @@
+Adding New Checks
+=================
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+When adding new violations to edx-lint, there currently isn't clarity around the following:
+
+* What changes should be considered a breaking change, requiring a major version bump.
+* How to avoid too avoid many breaking changes.
+* When and where to best communicate changes.
+* How to ensure that new violations are being picked up by repositories.
+
+Decisions
+---------
+
+* New checks should be introduced as optional, even if you ultimately want them to be required by default.
```

Comment:
Am I forgetting something? What's an optional pylint check? Or is this for checks beyond pylint?

Reply:
I wasn't sure what I meant when I first wrote this.  What I now think is that checks will be disabled by default (i.e. optional, or opt-in using a local override to enable), and eventually will be enabled by default (i.e. required by default, or opt-out using a local override to disable).

First, can you confirm that this is a reasonable plan? Also, is it more clear to lead with pylint language like "enabled/disabled" vs "required/optional"? Should I drop the language of "required/optional" altogether?

Reply:
I'm not sure this is possible without modifying the pylintrc file for every project. This is because all linters from edx_lint.pylint are run by default.

Reply:
@regisb @nedbat: Thanks for your feedback. A combo of your notes, and doing some research, helped me understand what is going on.
1. I simplified the docs, because in most cases, we are stuck with breaking changes.
2. I added some more vague notes about optional plugins, because it is an option, but I don't have enough experience with this to recommend it, etc.